Update changelog, prepare `15-4.2.4` release (#227)
pganalyze/libpg_query: CHANGELOG.md

All versions are tagged by the major Postgres version, plus an individual semver for this library itself.

## 15-4.2.4   2023-12-20

> * Add Token `ASCII_36` [#211](https://github.com/pganalyze/libpg_query/pull/211), [#219](https://github.com/pganalyze/libpg_query/pull/219)

Reviewer: ```suggestion
* Scanner: Add token `ASCII_36` ("$") to support queries like "SELECT $identifier" [#211](https://github.com/pganalyze/libpg_query/pull/211), [#219](https://github.com/pganalyze/libpg_query/pull/219)
  - Whilst these queries are not valid SQL and would fail parsing, this token can show up when using `pg_query_scan` or `pg_query_split_with_scanner` directly
```

> * Fix failing type cast in WebAssembly builds [#223](https://github.com/pganalyze/libpg_query/pull/223)

Reviewer: ```suggestion
* Normalize: Fix incorrect type cast [#223](https://github.com/pganalyze/libpg_query/pull/223)
```

(I don't think its a WebAssembly specific issue, just that it surfaced there -- its not clear why this worked otherwise)

* Deparser:
  - Fix some `BooleanTest` cases [#206](https://github.com/pganalyze/libpg_query/issues/206)
  - Fix `ALTER TABLE ... ENABLE TRIGGER ALL` [#222](https://github.com/pganalyze/libpg_query/pull/222)
  - Add parens around type casts in `IndexElem` [#214](https://github.com/pganalyze/libpg_query/pull/214)
  - Treat type casts in `func_expr_windowless` as functions [#214](https://github.com/pganalyze/libpg_query/pull/214)

> * Support passing parse mode and additional parse options [#216](https://github.com/pganalyze/libpg_query/pull/216)

Reviewer: ```suggestion
* Support changing parse mode and config settings affecting the parser [#216](https://github.com/pganalyze/libpg_query/pull/216)
```

>   - Add additional parse modes to support parsing PL/pgSQL

Reviewer: ```suggestion
  - Alternate parse modes are useful for parsing PL/pgSQL expressions, as well as type names
```

>   - Flags to disable various string warnings

Reviewer: ```suggestion
  - Additionally, you can now change config settings that affect parsing, like `standard_conforming_strings`
  - To pass options, use the new methods ending in `_opts`, e.g. `pg_query_parse_opts`
```
